src/network/networkctl.c

```diff
@@ -242,6 +249,64 @@ static const LinkInfo* link_info_array_free(LinkInfo *array) {
 }
 DEFINE_TRIVIAL_CLEANUP_FUNC(LinkInfo*, link_info_array_free);
 
+DEFINE_TRIVIAL_CLEANUP_FUNC(CURL*, curl_easy_cleanup);
+DEFINE_TRIVIAL_CLEANUP_FUNC(struct curl_slist*, curl_slist_free_all);
+
+static int acquire_azure_cloud_metadata(NetworkCloudManager **ret) {
```

Comment:
this function should move to the new file as well

Reply:
For networkd it will be callback and we won't so a sync

Reply:
Sorry, I don't follow, why wouldn't moving this and link_dump_azure_cloud_public_ip to network-cloud-azure.c work?

Reply:
It will work I just want to avoid
1.  Not to put the whole share lib into curl dependency and libsystemd-network
2. 

```json_parse``` will be in callback function in networkd which will be async. Only the curl code will be same which is few.

Reply:
IMHO for libsystemd-networkd it should be fine as a new dependency. As long as the files are moved out of libsystemd-shared (where yeah it would be a problem) it sounds ok.

To keep it optional, you can wrap the body of acquire_azure_cloud_metadata into ```#if HAVE_LIBCURL``` and have an \#else with an empty ```return -EOPNOTSUPP```.

Reply:
yeah I am ok with libsystemd. Only the curl init and set functions. Json will not be there.